pkg/kube/controllers/quarksstatefulset/active_passive_reconciler.go

```diff
@@ -0,0 +1,352 @@
+			return reconcile.Result{}, nil
+		}
+		// Reconcile failed due to error - requeue
+		ctxlog.Errorf(ctx, "Failed to get quarks statefulset '%s': %s", request.NamespacedName, err)
+		return reconcile.Result{}, err
+	}
+
+	statefulSetVersions, err := r.listStatefulSetVersions(ctx, qSts)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	maxAvailableVersion := qSts.GetMaxAvailableVersion(statefulSetVersions)
+
+	statefulSets, err := listStatefulSetsFromInformer(ctx, r.client, qSts)
+	if err != nil {
+		// Reconcile failed due to error - requeue
+		return reconcile.Result{}, errors.Wrapf(err, "couldn't list StatefulSets for active/passive reconciliation.")
+	}
+
+	// Retrieve the last versioned statefulset, based on the max available version
+	desiredSts, err := getDesiredSts(statefulSets, fmt.Sprintf("%d", maxAvailableVersion))
+	if err != nil {
+		// Reconcile failed due to error - requeue
+		return reconcile.Result{}, errors.Wrapf(err, "couldn´t get the last versioned statefulset owned by QuarksStatefulSet")
+	}
+
+	// ctxlog.Info(ctx, "Max available version for quarks sts'", qSts.Name, "' is version '", maxAvailableVersion, "' in namespace '", qSts.Namespace, "'.")
+	ownedPods, err := r.getStsPodList(ctx, desiredSts)
+	if err != nil {
+		// Reconcile failed due to error - requeue
+		return reconcile.Result{}, errors.Wrapf(err, "couldn't retrieve pod items from sts: %s", desiredSts.Name)
+	}
+
+	// retrieves the ActivePassiveProbe children key,
+	// this is the container name in where the ActivePassiveProbe
+	// cmd, needs to be executed
+	containerName, err := getProbeContainerName(qSts.Spec.ActivePassiveProbe)
+	if err != nil {
+		// Reconcile failed due to error - requeue
+		return reconcile.Result{}, errors.Wrapf(err, "None container name found in probe for %s QuarksStatefulSet", qSts.Name)
+	}
+
+	err = r.defineActiveContainer(ctx, containerName, ownedPods, qSts)
+	if err != nil {
+		// Reconcile failed due to error - requeue
+		return reconcile.Result{}, err
+	}
+
+	periodSeconds := time.Second * time.Duration(qSts.Spec.ActivePassiveProbe[containerName].PeriodSeconds)
+	if periodSeconds == (time.Second * time.Duration(0)) {
+		ctxlog.WithEvent(qSts, "active-passive").Debugf(ctx, "periodSeconds probe was not specified, going to default to 10 secs")
+		periodSeconds = time.Second * 5
+	}
+
+	// Reconcile for any reason than error after the ActivePassiveProbe PeriodSeconds
+	return reconcile.Result{RequeueAfter: periodSeconds}, nil
+}
+
+func (r *ReconcileStatefulSetActivePassive) defineActiveContainer(ctx context.Context, container string, pods *corev1.PodList, qSts *qstsv1a1.QuarksStatefulSet) (err error) {
+	var p *corev1.Pod
+	labelledPods := r.getActiveLabels(pods, container)
+
+	probeCmd := qSts.Spec.ActivePassiveProbe[container].Exec.Command
+
+	// switch depending on pods with active label
+	switch pl := len(labelledPods); {
+	// Nothing have been labeled yet
+	case pl == 0:
```

Comment:
we should always check all pods - more than one pod could be active

Reply:
Yes, I added support for this

Reply:
You're still only making on pod active at once.
I see you're always making sure there's only one active pod at the same time.

I don't think we need to do this - we should just run the probe and mark accordingly (for all pods, all the time).